pkg/querier/querier.go

```diff
@@ -46,16 +49,32 @@ func New(cfg Config, clientCfg client.Config, ring ring.ReadRing, store chunk.St
 	}, nil
 }
 
+type responseFromIngesters struct {
+	addr     string
+	response interface{}
+}
+
 // forAllIngesters runs f, in parallel, for all ingesters
 // TODO taken from Cortex, see if we can refactor out an usable interface.
-func (q *Querier) forAllIngesters(f func(logproto.QuerierClient) (interface{}, error)) ([]interface{}, error) {
+func (q *Querier) forAllIngesters(f func(logproto.QuerierClient) (interface{}, error)) ([]responseFromIngesters, error) {
 	replicationSet, err := q.ring.GetAll()
 	if err != nil {
 		return nil, err
 	}
 
-	resps, errs := make(chan interface{}), make(chan error)
-	for _, ingester := range replicationSet.Ingesters {
+	return q.forGivenIngesters(replicationSet.Ingesters, f)
+}
+
+// forGivenIngesters runs f, in parallel, for given ingesters
+// TODO taken from Cortex, see if we can refactor out an usable interface.
+func (q *Querier) forGivenIngesters(ingesters []ring.IngesterDesc, f func(logproto.QuerierClient) (interface{}, error)) ([]responseFromIngesters, error) {
+	replicationSet, err := q.ring.GetAll()
```

Comment:
I don't think we should be using `replicationSet` here. We should be using the ingesters passed down.